specs/DesignDiscussions/InputLabelsOptionLabel.md

```diff
@@ -0,0 +1,440 @@
+* isVisible: labels should always be visible as long as the associated input is
+* height: input elements should control the height for the whole set of label, input and error message 
+
+Allowing the control of this styling via support of native styling on required and optional labels (i.e. CSS, Xaml Styles, etc) may also be required but will be set out of scope for v1 of this feature. This decision speaks to a larger discussion of the direction of host config that is beyond the scope of this document.
+
+#### Required and Optional indicators
+
+ By default, we will mark required inputs with a `*`. We may want, however, to provide the option for the host to configure a suffix to the label for required or optional inputs. This allows the host to add a `*` to required labels, or to add the words "required" or "optional". Consider a host config similar to the below:
+
+  ```json
+"inputLabelFormatting": 
+{
+	"requiredInputs": {
+		"suffix":"*",
+		"weight": "Bolder",
+	},
+	"optionalInputs": {
+		"suffix":"(optional)",
+		"isSubtle":true
+	}
+}
+```
+If we want even more detailed control, we could allow the host to specify something like `{labelName} (optional)` or `(optional) {labelName}` in order to allow them to position their indicators as they wish. We could make use of the common expression language being used for templating to support this functionality, but that would require taking a dependency on the common expression language implementation that would not otherwise be required for hosts not using templating.
+
+> One concern with allowing hosts to specify arbitrary label templates or suffixes, is that it opens up a localization issue.  Whatever word they provide should be localized into the language of the card. If we introduce this functionality, we should make sure we have a localization story that covers this scenario. 
+
+> When using these host properties, we would need to only apply them if all inputs on the card have labels so that required fields are marked consistently. We don't want to end up in a situation where some inputs in the card are marked as required in the label and others on the field itself, as this may lead to a confusing user experience.
+
+### Card Author Formatting
+
+We may wish to prioritize Host formatting over card author formatting in order to allow our hosts to impose a specified look and feel to forms displayed in their app. That said, if we don't allow card authors to specify formatting of labels, those authors who wish to do so will simply opt-out of our label approach and revert to using the existing unassociated TextBlocks and RichTextBlocks. This has accessibility and usability impact on our end customers, as they won't have access to the improved experiences made possible when our renderers have semantic knowledge of which text blocks are the labels for a given input.
```

Comment:
I think this is a fair point. But having labels without support for card author formatting in v1 is acceptable as long as we document and provide guidance samples etc on building an Accessible Card with the current AdaptiveCards stack, and then iterate on our feature set after shipping the basic feature.

Reply:
I think accessibility wise, most of the work will be done through rendering, as long as the user has enough information on the input they should be able to work through it. I agree that we should provide guidance on how to make accessible cards.